### ISSUE TITLE
Error / loop failed work / transaction

### ISSUE THREAD

**Issue by drewbs-b** [iExecBlockchainComputing/public-pool-scripts#45]
Woke up this morning to check my worker. Was stuck in a loop as seen below. Not sure if the log is still available if so I can check it later, but when I scrolled up there was a task assigned and some errors involved. Restarted worker before copying and pasting that portion of the log. If needed and there's a way to get it I can go back and find the additional data from that log. There was a transaction associated with the job, with fees paid from my account.

Transaction: https://etherscan.io/tx/0x563556242277ac4c3c746da5ca30d6aee41e0494dc692e4529ba523d15cd8ff0
My wallet: 
https://etherscan.io/address/0xacdba7adae48167c130c2fbdb08b69db135b453c

Restarting the worker got things back to normal.

19/Dec/2018:11:53:55 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Ping = 206 ms
[19/Dec/2018:11:53:56 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Alive period from server = 60
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.synchronize(ThreadAlive.java:439)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:136)
[19/Dec/2018:11:53:56 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:54:11 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:54:26 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
0x031ee1c8000000000000000000000000ccb4c799749cf8b5a1cce4b3e813d335e55ea93da7532de671cd1dcb7a256f06a54c2175db7447b31622f28b492c3dc103c11e65aea34693f0b8fd4dd136a1f82b5310a6bb498b4cf70b5efb81a51ef3481d8091000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000[19/Dec/2018:11:54:41 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
11:54:56.399 [ThreadAlive] INFO  c.i.w.workerpool.WorkerPoolService - GetContributionModel [workOrderId:0xccb4c799749cf8b5a1cce4b3e813d335e55ea93d, transactionStatus:SUCCESS] 
==> about to get workOrderStatus
[19/Dec/2018:11:54:56 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
==> workOrderIsRevealing: false
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.checkJob(ThreadAlive.java:341)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:155)
[19/Dec/2018:11:54:57 +0000] [xtremweb.worker.ThreadLaunch_ThreadAlive_19] ERROR : getThreadByWorkUid() can't find work 0f60f208-eb07-4d1d-aa84-4cd8a940874e
[19/Dec/2018:11:54:57 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] WARN : workAlive(0f60f208-eb07-4d1d-aa84-4cd8a940874e) can't find working thread
[19/Dec/2018:11:54:57 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Ping = 213 ms
[19/Dec/2018:11:54:57 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Alive period from server = 60
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.synchronize(ThreadAlive.java:439)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:136)
[19/Dec/2018:11:55:11 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:55:26 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:55:41 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:55:56 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
11:55:57.688 [ThreadAlive] INFO  c.i.w.workerpool.WorkerPoolService - GetContributionModel [workOrderId:0xccb4c799749cf8b5a1cce4b3e813d335e55ea93d, transactionStatus:SUCCESS] 
==> about to get workOrderStatus
==> workOrderIsRevealing: false
[19/Dec/2018:11:55:59 +0000] [xtremweb.worker.ThreadLaunch_ThreadAlive_19] ERROR : getThreadByWorkUid() can't find work 0f60f208-eb07-4d1d-aa84-4cd8a940874e
[19/Dec/2018:11:55:59 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] WARN : workAlive(0f60f208-eb07-4d1d-aa84-4cd8a940874e) can't find working thread
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.checkJob(ThreadAlive.java:341)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:155)
[19/Dec/2018:11:55:59 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Ping = 210 ms
[19/Dec/2018:11:55:59 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Alive period from server = 60
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.synchronize(ThreadAlive.java:439)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:136)
[19/Dec/2018:11:56:11 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:56:26 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
^C[19/Dec/2018:11:56:41 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:56:56 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
11:56:59.768 [ThreadAlive] INFO  c.i.w.workerpool.WorkerPoolService - GetContributionModel [workOrderId:0xccb4c799749cf8b5a1cce4b3e813d335e55ea93d, transactionStatus:SUCCESS] 
==> about to get workOrderStatus
==> workOrderIsRevealing: false
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.checkJob(ThreadAlive.java:341)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:155)
[19/Dec/2018:11:57:00 +0000] [xtremweb.worker.ThreadLaunch_ThreadAlive_19] ERROR : getThreadByWorkUid() can't find work 0f60f208-eb07-4d1d-aa84-4cd8a940874e
[19/Dec/2018:11:57:00 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] WARN : workAlive(0f60f208-eb07-4d1d-aa84-4cd8a940874e) can't find working thread
[19/Dec/2018:11:57:00 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Ping = 210 ms
java.lang.Exception: Stack trace
	at java.lang.Thread.dumpStack(Thread.java:1336)
	at xtremweb.worker.PoolWork.removeWork(PoolWork.java:310)
	at xtremweb.worker.ThreadAlive.synchronize(ThreadAlive.java:439)
	at xtremweb.worker.ThreadAlive.run(ThreadAlive.java:136)
[19/Dec/2018:11:57:01 +0000] [xtremweb.worker.ThreadAlive_ThreadAlive_19] INFO : Alive period from server = 60
[19/Dec/2018:11:57:11 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:57:26 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST 
[19/Dec/2018:11:57:41 +0000] [xtremweb.worker.CommLL_CommManager_15] INFO : Added Communcation : WORKREQUEST

**Comment by branciard** [comment]
Indeed, your contribution was valid but the workorder do not reach the finilized state. I will manually reward you for this and give the manual transaction in this issue. 

We see the workorder is still active and not completed.

```

iexec work show 0xccb4c799749cf8b5a1cce4b3e813d335e55ea93d --chain mainnet
ℹ iExec SDK update available 2.2.39 →  2.3.1, Run "npm -g i iexec" to update

ℹ using chain [mainnet]
✔ work 0xccb4c799749cf8b5a1cce4b3e813d335e55ea93d status is ACTIVE, details:
m_workerpool:          0x0ed5e70874487bf21ee65343fb6c5fa13382cdb6
m_params:              {"cmdline":"monte-carlo normal"}
m_requester:           0x8ec966ff9752c5de46105155ef1e9a3a38542358
m_app:                 0xcb0bba8cf239e167af92a51c4212adbf9b479f6a
m_dataset:             0x0000000000000000000000000000000000000000
m_emitcost:            0
m_uri:
m_stdout:
m_resultCallbackProof: 0x0000000000000000000000000000000000000000000000000000000000000000
m_iexecHubAddress:     0x0d5ef019ca4c5cc413ee892ced89d7107c5f424d
m_callback:            0x0000000000000000000000000000000000000000
m_status:              1
m_marketorderIdx:      384
m_stderr:
m_beneficiary:         0x0000000000000000000000000000000000000000
m_statusName:          ACTIVE

ℹ if work is not "COMPLETED" after Thu Dec 20 2018 04:41:56 GMT+0100 (CET) you can claim the work to get a full refund using "iexec work claim"

```

**Comment by branciard** [comment]
23 RLC manual reward for worker 0xacdba7adae48167c130c2fbdb08b69db135b453c see tx : https://etherscan.io/tx/0x16de2a4b43cff5f670257b9a3fbd0b4bcc755abc738c6038477e481343b2b10e
